contracts/src/modules/Coverage.sol

```diff
@@ -59,28 +60,40 @@ contract Coverage is Ownable, ModuleBase {
         bytes calldata initData
     ) public override onlyRegistrar returns (uint256) {
         require(isWhitelisted[caller], "Not whitelisted");
+        require(_owner == address(this), "Risk fee not paid to pool"); // TODO registrar assumes that the owner is the one being paid
+        require(currency == usdc, "Incorrect currency");
+
+        uint256 maturityDate = block.timestamp + coveragePeriod;
+
+        if (poolStart == 0) {
+            poolStart = block.timestamp;
+            liquidityReleaseDate = block.timestamp + liquidityLockupPeriod;
+        } else {
+            require(maturityDate <= liquidityReleaseDate);
+        }
 
         (address holder, uint256 amount, uint256 riskScore) = abi.decode(
             initData,
             (address, uint256, uint256)
-        );
+        );  // TODO do we need the risk score right now?
 
         require(instant == (amount / 10000) * riskScore, "Risk fee not paid");
-        // TODO: maybe onramp should own nota? (currently the registrar assumes that the owner is the one being paid)
-        require(_owner == address(this), "Risk fee not paid to pool");
-        require(currency == usdc, "Incorrect currency");
+        
+        availableFunds -= amount;  // Throws on underflow (using pragma >0.8)
```

Comment:
This assumes that the pool is full collateralized but with the coverage model the pool is under-collateralized. For Stably we're planning to offer 200k of coverage backed by 10k in the pool. Maybe a have a variable to track the `reserveRatio`?